hack/changelog.py

```diff
@@ -0,0 +1,128 @@
+emojiBugfix = str('🐛')
+emojiDocs = str('📖')
+emojiInfra = str('🌱')
+emojiBreaking = str('⚠')
+
+
+class PRType(Enum):
+    UncategorizedPR = 1
+    BreakingPR = 2
+    FeaturePR = 3
+    BugfixPR = 4
+    DocsPR = 5
+    InfraPR = 6
+
+
+def pr_type_from_title(pr_title: str):
+    t = pr_title.strip()
+    if len(pr_title) == 0:
+        return PRType.UncategorizedPR, t
+
+    if pr_title.startswith(':sparkles:') or pr_title.startswith(emojiFeature):
+        return PRType.FeaturePR, t.removeprefix(':sparkles:').removeprefix(emojiFeature).strip()
+    elif pr_title.startswith(':bug:') or pr_title.startswith(emojiBugfix):
+        return PRType.BugfixPR, t.removeprefix(':bug:').removeprefix(emojiBugfix).strip()
+    elif pr_title.startswith(':book:') or pr_title.startswith(emojiDocs):
+        return PRType.DocsPR, t.removeprefix(':book:').removeprefix(emojiDocs).strip()
+    elif pr_title.startswith(':seedling:') or pr_title.startswith(emojiInfra):
+        return PRType.InfraPR, t.removeprefix(':seedling:').removeprefix(emojiInfra).strip()
+    elif pr_title.startswith(':warning:') or pr_title.startswith(emojiBreaking):
+        return PRType.BreakingPR, t.removeprefix(':warning:').removeprefix(emojiBreaking).strip()
+
+    return PRType.UncategorizedPR, t
+
+
+def change_entry(pr_title, number, author):
+    return "%s (#%d) @%s" % (pr_title, number, author)
+
+
+def section_if_present(changes: [], pr_title):
+    if len(changes) > 0:
+        print("")
+        print("## %s\n" % pr_title)
+        print("")
+        for change in changes:
+            print("- %s\n" % change)
+
+
+if __name__ == '__main__':
+    args = sys.argv[1:]
+    token = args[0]
+    if len(token) == 0:
+        token = os.getenv("GITHUB_TOKEN")
+    auth = Auth.Token(token)
+    repo_name = args[1]
+    release_tag = args[2]
+    g = Github(auth=auth)
+    repo = g.get_repo("open-cluster-management-io/addon-contrib")
+    pulls = repo.get_pulls(state='closed', sort='created', base='main', direction='desc')
+
+    # get the last release tag
+    tags = repo.get_tags()
+    if tags.totalCount == 0:
+        print("no tags in the repo")
+        sys.exit()
+    elif tags.totalCount == 1:
+        last_release_tag = tags[0].name
+    else:
+        last_release_tag = tags[1].name
+
+    # get related PR from the last release tag
+    last_release_pr = 0
+    release_word = "in"
+
+    if tags.totalCount > 1:
+        release_word = "since"
+        for tag in tags:
+            if tag.name == last_release_tag:
+                tag_pulls = tag.commit.get_pulls()
+                last_release_pr = tag_pulls[0].number
+                break
+
+    # collect all PRs from the last release tag
+    features = []
+    bugs = []
+    breakings = []
+    docs = []
+    infras = []
+    uncategorized = []
+    for pr in pulls:
+        if pr.number == last_release_pr:
+            break
+        if pr.is_merged():
+            if repo_name not in [l.name for l in pr.labels]:
```

Comment:
@qiujian16 it's copied from lab repo, it added a label filter here, is this what we want?